p2p/exchange.go

```diff
@@ -47,6 +47,7 @@ func NewExchange[H header.Header](
 	host host.Host,
 	peers peer.IDSlice,
 	connGater *conngater.BasicConnectionGater,
+	tracker *PeerTracker,
```

Comment:
So this PR is breaking... For node and Rollkit. 
Are we 100% we can't pass the PeerIDStore here as an option and need to make the tracker public? 
I'm not too fond of that.

Reply:
>  it would be ugly to pass peerIDStore to exchange just so it can pass to peer tracker on construction.

I don't see any problem with this. We already do that with the host for example. I don't see any fundamental problems with this

Reply:
So -  of course we can keep the tracker private and extend params for Exchange to include ability to pass a whole PeerIDStore in there, but the reason why I was in favour of exporting tracker anyway is bc i think it may eventually feed into information of some other component or need to have some sort of public api for another broader peer management system to use. This is just a feeling atm so I'm not married to it, but I also find it strange to pass in all the components to exchange constructor just for exchange constructor to construct tracker there - makes more sense to construct the peertracker as desired and then pass it to exchange.

So yes, this is breaking, but I think we'll eventually create an API around it somehow. 

I'm happy to revert it, I just think separating the constructor out and passing in tracker felt cleaner.

Reply:
Agreed to revert in sync

Reply:
done @Wondertan